src/registry/tests.rs

```diff
@@ -0,0 +1,111 @@
+//! Tests for the registry
+
+use std::collections::BTreeMap;
+use std::sync::mpsc;
+use std::thread;
+
+use super::*;
+
+use rustc_serialize::{Decodable, Decoder};
+use rustc_serialize::json;
+use rustc_serialize::json::{Json, ToJson};
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+struct Point {
+    x: i32,
+    y: i32
+}
+
+impl ToJson for Point {
+    fn to_json(&self) -> Json {
+        let mut map = BTreeMap::new();
+        map.insert("x".to_string(), Json::I64(self.x as i64));
+        map.insert("y".to_string(), Json::I64(self.y as i64));
+        Json::Object(map)
```

Comment:
Why does the `Json` library make this have to be a BTreeMap? Seems like an odd data structure to use for this, why not just anything that utilizes the `Entry` trait?